search/mongo_search.go

```diff
@@ -829,10 +832,46 @@ func (m *MongoSearcher) createNumberQueryObject(n *NumberParam) bson.M {
 	single := func(p SearchParamPath) bson.M {
 		l, _ := n.Number.RangeLowIncl().Float64()
 		h, _ := n.Number.RangeHighExcl().Float64()
-		return buildBSON(p.Path, bson.M{
-			"$gte": l,
-			"$lt":  h,
-		})
+		exact, _ := n.Number.Value.Float64()
+
+		var criteria bson.M
+
+		switch n.Prefix {
+		case EQ:
+			// Equality is in the range [l, h)
+			criteria = bson.M{
+				"$gte": l,
+				"$lt":  h,
+			}
+		case NE:
+			// In the range (-inf, l) || [h, inf)
+			criteria = bson.M{
+				"$or": []bson.M{
+					bson.M{"$lt": l},
+					bson.M{"$gte": h},
+				},
+			}
+		case GT:
+			criteria = bson.M{
+				"$gt": exact,
+			}
+		case LT:
+			criteria = bson.M{
+				"$lt": exact,
+			}
+		case GE:
+			criteria = bson.M{
+				"$gte": exact,
+			}
+		case LE:
+			criteria = bson.M{
+				"$lte": exact,
+			}
```

Comment:
There are some potential logical inconsistencies here -- mainly from the implementation of EQ.  For example, it's possible for a number to be _both_ EQ to and LT another number at the same time.  Similarly, it's possible for a number to be EQ to another number but _not_ LE to the same number...

I think using `exact` for GT and LT probably makes sense, but I wonder if you should use `l` for GE and `h` for LE...?